### ISSUE TITLE
Evaluate based on Train/Test Split

### ISSUE THREAD

**Issue by rgstephens** [RasaHQ/rasa_nlu#1625]
**Rasa NLU version**: 0.13.0

**Enhancement**:

I've started working on an enhancement to evaluate that uses the `train_test_split` function and does an evaluation of the model based upon a dynamically created test set that is pulled from the training data. You can then create a confusion matrix and errors.json like the normal evaluate.

I'm working with the following syntax for evaluate:

```
python rasa_nlu.evaluate.py --mode traintest --config nlu_config.yml --data training/ --pct 90 --cutoff 5
```

Is this of interest as a general enhancement? If so, I can create a PR?

**Comment by akelad** [comment]
Yes we would be for sure! I think maybe rather than having an additional mode though, it might make sense to add it as a flag to the standard evaluate mode?/

**Comment by rgstephens** [comment]
Okay, so how about if the `--pct` option triggers the train/test split path:

```
python rasa_nlu.evaluate.py --mode evaluation --config nlu_config.yml --data training/ --pct 90 --cutoff 5
```

**Comment by akelad** [comment]
yeah sure, something like that makes sense :)

**Comment by veronicaads** [comment]
hello, can i ask about the same ? i have tested my models with seperated data test and i want get the confusion matrix plot 
i run this script
python -m rasa_nlu.evaluate \
> --data data/examples/evaluate_data.json \
> --model models/nlu/default/chat \
> --confmat=data\examples

and

python -m rasa_nlu.evaluate \
> --data data/examples/evaluate_data.json \
> --model models/nlu/default/chat \
> --confmat data\examples

but why the plot isnt at the directory , btw i run at hybrid cloud server

thankyou

**Comment by akelad** [comment]
see comment in #1632
